app/lib/timetable/timetable_add_event/timetable_add_event_dialog.dart

```diff
@@ -75,27 +75,39 @@ class TimetableAddEventDialog extends StatelessWidget {
   final bool isExam;
   late AddEventDialogController? controller;
   late final FocusNode titleFocusNode;
-  bool _hasRequestedTitleFocus = false;
-
   static const tag = "timetable-event-dialog";
 
+  @override
+  State<TimetableAddEventDialog> createState() =>
+      _TimetableAddEventDialogState();
+}
+
+class _TimetableAddEventDialogState extends State<TimetableAddEventDialog> {
+  late AddEventDialogController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = widget.controller ??
+        AddEventDialogController(
+          isExam: widget.isExam,
+          api: EventDialogApi(BlocProvider.of<SharezoneContext>(context).api),
+          markdownAnalytics: BlocProvider.of<MarkdownAnalytics>(context),
+        );
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      delayKeyboard(context: context, focusNode: widget.titleFocusNode);
+    });
+  }
+
   bool hasModifiedData() {
-    return controller!.title.isNotEmpty || controller!.description.isNotEmpty;
+    return controller.title.isNotEmpty || controller.description.isNotEmpty;
   }
 
   @override
   Widget build(BuildContext context) {
-    if (!_hasRequestedTitleFocus) {
-      _hasRequestedTitleFocus = true;
-      delayKeyboard(context: context, focusNode: titleFocusNode);
```

Comment:
@Jonas-Sander Do you know if there was a reason why you called the `delayKeyboard` in the `build()` method instead of using a `StatefulWidget` with `initState`?

Reply:
No, probably just laziness.